lib/setup/setupInstall.js

```diff
@@ -1525,62 +1530,108 @@ function Install(options) {
+
+                if (params.custom) {
+                    // remove adapter from custom
+                    await that.removeCustomFromObjects([...instances, adapter]);
+                }
+
+                await uninstallNpm();
             }
-        });
+        } catch (e) {
+            console.error(`There was an error uninstalling ${adapter} on ${hostname}: ${e.message}`);
+        }
+        return tools.maybeCallback(callback, adapter, resultCode);
     };
 
-    this.deleteInstance = function (adapter, instance, callback) {
-        const knownObjectIDs  = [];
-        const knownStateIDs   = [];
-
-        return that.enumerateAdapterInstances(knownObjectIDs, null, adapter, instance)
-            .then(() => that.enumerateAdapterDevices(knownObjectIDs, adapter, instance))
-            .then(() => that.enumerateAdapterChannels(knownObjectIDs, adapter, instance))
-            .then(() => that.enumerateAdapterStateObjects(knownObjectIDs, adapter, instance))
-            .then(() => that.enumerateAdapterStates(knownStateIDs, adapter, instance))
-            .then(() => that.enumerateAdapterDocs(knownObjectIDs, adapter, instance))
-            .then(() => that.deleteAdapterObjects(knownObjectIDs))
-            .then(() => that.deleteAdapterStates(knownStateIDs))
-            .then(() => callback && callback(adapter, instance))
-        ;
+    this.deleteInstance = async (adapter, instance, callback) => {
+        const knownObjectIDs = [];
+        const knownStateIDs = [];
+
+        await that.enumerateAdapterInstances(knownObjectIDs, null, adapter, instance);
+        await that.enumerateAdapterDevices(knownObjectIDs, adapter, instance);
+        await that.enumerateAdapterChannels(knownObjectIDs, adapter, instance);
+        await that.enumerateAdapterStateObjects(knownObjectIDs, adapter, instance);
+        await that.enumerateAdapterStates(knownStateIDs, adapter, instance);
+        await that.enumerateAdapterDocs(knownObjectIDs, adapter, instance);
+        await that.deleteAdapterObjects(knownObjectIDs);
+        await that.deleteAdapterStates(knownStateIDs);
+        if (params.custom) {
+            // delete instance from custom
+            await that.removeCustomFromObjects([`${adapter}.${instance}`]);
+        }
+        return tools.maybeCallback(callback, adapter, instance);
 
         // TODO delete meta objects - i think a recursive deletion of all child object would be less effort.
     };
 
+    /**
+     * Removes the custom attribute of the provided adapter/instance
+     *
+     * @param {string[]} ids - id of the adapter/instance to check for
+     * @returns {Promise<void>}
+     */
+    this.removeCustomFromObjects = async ids => {
+        // get all objects which have a custom attribute
+        const res = await objects.getObjectViewAsync('system', 'custom', {
+            startkey: '',
+            endkey: '\u9999'
+        });
+
+        for (const row of res.rows) {
```

Comment:
Please add a check that res.rows is there ... else we could get cases like in https://github.com/ioBroker/ioBroker.js-controller/issues/1428 ... or are we sure that it do not happens? :-)